src/main/java/seedu/address/model/trip/Trip.java

```diff
@@ -0,0 +1,151 @@
+package seedu.address.model.trip;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.driver.Driver;
+import seedu.address.model.person.passenger.Passenger;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Trip.
+ * Guarantees: details relevant for a trip are present and not null, field values are validated, immutable.
+ */
+public class Trip {
+
+    // Data fields
+    private final TripDay tripDay;
+    private final TripTime tripTime;
+    private final Set<Passenger> passengers = new HashSet<>();
+    private final Set<Tag> tags = new HashSet<>();
+    private Driver driver;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Trip(Driver driver, TripDay tripDay, TripTime tripTime, Set<Passenger> passengers, Set<Tag> tags) {
+        requireAllNonNull(driver, tripDay, tripTime);
+
+        // ensures a trip cannot be created with no passengers
+        if (passengers.isEmpty()) {
+            throw new NullPointerException();
```

Comment:
NullPointerException is probably the wrong exception to throw

Reply:
You're right. I was intending to throw the same exception that the requireAllNotNull() check in the line above it does but an illegal argument exception would be more fitting.